clients/mobile/src/LocationService/LocationService.ts

```diff
@@ -40,4 +63,84 @@ export class LocationService {
       }
     }
   }
+
+  async preferedLocationAutocomplete(location: string, city: string): Promise<Array<{ label: string; value: string }>> {
+    const COUNTRY = 'BD'
+
+    try {
+      const response = await this.httpClient.get<{ predictions: Prediction[] }>('/place/autocomplete/json', {
+        input: location,
+        types: 'geocode',
+        components: `country:${COUNTRY}`,
+        key: GOOGLE_MAP_API_KEY
+      })
+      const formattedData = response.predictions
```

Comment:
maybe
```ts
const formattedData = response.predictions.reduce((acc: Array<{ label: string; value: string }>, prediction: Prediction) => {
  if (
    prediction.types.includes('neighborhood') ||
    prediction.types.includes('political')
  ) {
    acc.push({
      label: prediction.description,
      value: prediction.description
    });
  }
  return acc;
}, []);

```

Reply:
Create a separate function for formatting and filtering 

```
filterAndFormatPredictions(predictions: Prediction[], allowedTypes: string[]): Array<{ label: string; value: string }> {
    return predictions.reduce((acc: Array<{ label: string; value: string }>, prediction) => {
      if (prediction.types.some((type) => allowedTypes.includes(type))) {
        acc.push({
          label: prediction.description,
          value: prediction.description
        })
      }
      return acc
    }, [])
  }
```